### PR TITLE
New feature metadata edition

### LINKED ISSUE #3735: Create new "feature metadata edition" dialog
screen shot 2015-05-22 at 10 56 29

### LINKED ISSUE #3868: Different radio button styles
screen shot 2015-05-29 at 15 52 05
screen shot 2015-05-29 at 15 51 46

cc @saleiva @viddo

### PR DESCRIPTION
**Pull request by xavijam**
another-gif

---

It will let the user to:
- edit the data of any feature/row of the data.
- add a new column.
- change the column type.
- change the name of the column.

---

And, changed several edit fields views:
- Number field.
- Date field.
- Boolean field.
- String field.

Fixes #3735.

---

Also we have changed the behaviour of NASA radio buttons, fixes #3868.
nasa-search

### CODE REVIEW

File: app/assets/stylesheets/common/datepicker.css.scss [hunk 1]
```diff
@@ -23,9 +23,9 @@ $sDates-defaultWidth: 434px;
   height: 38px;
   padding: 0 $sMargin-elementInline;
   outline: none;
-  border: 1px solid #CCC;
-  border-radius: $border-radius;
-  background: white;
+  // border: 1px solid #CCC;
+  // border-radius: $border-radius;
+  // background: white;
```
Comment (viddo) [inline app/assets/stylesheets/common/datepicker.css.scss]: Remove if not going to be used

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4218/))

**Comment by saleiva** [comment]
bump!

**Comment by viddo** [comment]
@xavijam is it ready for review now?

**Comment by xavijam** [comment]
Yep, let me merge master in the meantime

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4280/))

File: lib/assets/javascripts/cartodb/common/dialogs/feature_data/feature_data_dialog_view.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+/**
+ *  Feature data edition dialog
+ */
+module.exports = BaseDialog.extend({
+
+  events: function() {
+    return _.extend({}, BaseDialog.prototype.events, {
+      'click .js-back': '_showForm'
+    });
+  },
```
Comment (viddo) [inline lib/assets/javascripts/cartodb/common/dialogs/feature_data/feature_data_dialog_view.js]: For the future, found a convenient method for extending events --> `BaseDialog.extendEvents({ … })`

File: lib/assets/javascripts/cartodb/common/dialogs/feature_data/form_view.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  },
+
+  _generateCollection: function() {
+    var self = this;
+    var tableSchema = this.table.get('schema');
+    var hiddenColumns = this.table.hiddenColumns;
+
+    _.each(tableSchema, function(pair) {
+      if (!_.contains(hiddenColumns, pair[0])) {
+        var mdl = self._generateModel(pair[0], pair[1], self.row.get(pair[0]));
```
Comment (viddo) [inline lib/assets/javascripts/cartodb/common/dialogs/feature_data/form_view.js]: This kind of code of handling a raw column data is used in a lot of places (as well as the merge modal I'm currently working on), why is it that we don't have a proper model to not have to do this manual fiddling in each place? 

I'm not saying that I expected it to be changed in this PR, but I wanted to point it out, I'd like to discuss it at some point. :)

File: lib/assets/javascripts/cartodb/common/edit_fields/edit_field_view.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    return this;
+  },
+
+  _initBinds: function() {
+    this.model.bind('error valid', this._setFieldStyle, this);
+    this.model.bind('change:readOnly', this.render, this);
+  },
+
+  _setFieldStyle: function() {
+    this.$el[ this.model.validationError ? 'addClass' : 'removeClass' ]('is-invalid');
```
Comment (viddo) [inline lib/assets/javascripts/cartodb/common/edit_fields/edit_field_view.js]: I found another useful method recently that could make this kind of code more succinct, `this.$el.toggleClass('is-invalid', this.model.validationError)`

**Comment by viddo** [comment]
Added a few comments on some details I saw, but overall looks OK. Judging by the location of the edit-fields stuff (common) I make the guess it's intended to be reused further ahead too?

**Comment by xavijam** [comment]
Yes, they will be used in other places, like here:

screen shot 2015-06-11 at 19 34 37

**Comment by viddo** [comment]
:+1:

**Comment by saleiva** [comment]
vamos nenos!

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4313/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4336/))

**Pull request merged**